### PR TITLE
add controllers for MovieSession, User and Movie, DTOs added

### PR DESCRIPTION
**Pull request by KovalAlexey**

### CODE REVIEW

File: src/main/java/com/dev/cinema/controllers/MovieController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    @GetMapping("/movies")
+    public List<MovieResponseDto> getAllMovies() {
+        return movieService.getAll().stream()
+                .map(movieMapper::toMovieDto)
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping("/add-movie")
```
Comment (margarita-ignatieva) [inline src/main/java/com/dev/cinema/controllers/MovieController.java]: guess POST has the same mapping as get: Add a movie - POST: /movies (from HW)

File: src/main/java/com/dev/cinema/service/mapper/MovieMapper.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import com.dev.cinema.model.Movie;
+import com.dev.cinema.model.dto.movie.MovieRequestDto;
+import com.dev.cinema.model.dto.movie.MovieResponseDto;
+import com.dev.cinema.service.MovieService;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MovieMapper {
+    private final MovieService movieService;
+
```
Comment (margarita-ignatieva) [inline src/main/java/com/dev/cinema/service/mapper/MovieMapper.java]: do you use movie service somewhere in the MovieMapper?

**Review by margarita-ignatieva — APPROVED** [review]

File: src/main/java/com/dev/cinema/service/mapper/MovieMapper.java (hunk 2 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/service/mapper/MovieMapper.java]: thanks, removing done

File: src/main/java/com/dev/cinema/controllers/MovieController.java (hunk 1 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/controllers/MovieController.java]: it's working fine (I tested with postman). But naming convention... thanks, just changed

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  MovieSessionMapper sessionMapper) {
+        this.movieSessionService = movieSessionService;
+        this.sessionMapper = sessionMapper;
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAllSessions(@RequestParam Long id,
+                                                        @RequestParam String showTime) {
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/controllers/MovieSessionController.java]: let's change String to LocalDate

File: src/main/java/com/dev/cinema/model/dto/UserResponseDto.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package com.dev.cinema.model.dto;
+
+public class UserResponseDto {
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/model/dto/UserResponseDto.java]:
```suggestion
public class UserRequestDto {
```

File: src/main/java/com/dev/cinema/model/dto/movie/MovieResponseDto.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package com.dev.cinema.model.dto.movie;
+
+public class MovieResponseDto {
+    private Long id;
+    private String title;
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/model/dto/movie/MovieResponseDto.java:5]: we can add description too

File: src/main/java/com/dev/cinema/model/dto/session/MovieSessionResponseDto.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
+package com.dev.cinema.model.dto.session;
+
+public class MovieSessionResponseDto {
+    private Long movieSessionId;
+    private String showTime;
+    private String movieTitle;
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/model/dto/session/MovieSessionResponseDto.java:6]: let's add movie id field

File: src/main/java/com/dev/cinema/service/mapper/MovieSessionMapper.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    public MovieSession toMovieSessionEntity(MovieSessionRequestDto movieSessionRequestDto) {
+        MovieSession movieSession = new MovieSession();
+        movieSession.setMovie(movieService.getById(movieSessionRequestDto.getMovieId()));
+        movieSession.setCinemaHall(cinemaHallService
+                .getById(movieSessionRequestDto.getCinemaHallId()));
+        movieSession.setShowTime(LocalDateTime.parse(movieSessionRequestDto.getShowTime()));
+        return movieSession;
+
+    }
+
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/service/mapper/MovieSessionMapper.java]: and here one empty line

File: src/main/java/com/dev/cinema/service/mapper/MovieSessionMapper.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    public MovieSession toMovieSessionEntity(MovieSessionRequestDto movieSessionRequestDto) {
+        MovieSession movieSession = new MovieSession();
+        movieSession.setMovie(movieService.getById(movieSessionRequestDto.getMovieId()));
+        movieSession.setCinemaHall(cinemaHallService
+                .getById(movieSessionRequestDto.getCinemaHallId()));
+        movieSession.setShowTime(LocalDateTime.parse(movieSessionRequestDto.getShowTime()));
+        return movieSession;
+
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/service/mapper/MovieSessionMapper.java]: remove empty line

File: src/main/java/com/dev/cinema/model/dto/UserResponseDto.java (hunk 4 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/model/dto/UserResponseDto.java]: done

File: src/main/java/com/dev/cinema/service/mapper/MovieSessionMapper.java (hunk 7 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/service/mapper/MovieSessionMapper.java]: done

File: src/main/java/com/dev/cinema/service/mapper/MovieSessionMapper.java (hunk 8 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/service/mapper/MovieSessionMapper.java]: done

File: src/main/java/com/dev/cinema/model/dto/session/MovieSessionResponseDto.java (hunk 6 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/model/dto/session/MovieSessionResponseDto.java:6]: added

File: src/main/java/com/dev/cinema/model/dto/movie/MovieResponseDto.java (hunk 5 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/model/dto/movie/MovieResponseDto.java:5]: added

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java (hunk 3 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/controllers/MovieSessionController.java]: changed

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**